src/coprocessor/codec/mysql/time/mod.rs

```diff
@@ -339,12 +342,12 @@ impl Time {
         Time::new(t, tp, fsp as i8)
     }
 
-    pub fn from_duration(tz: &FixedOffset, tp: u8, d: &MyDuration) -> Result<Time> {
+    pub fn from_duration(tz: &Tz, tp: u8, d: &MyDuration) -> Result<Time> {
         let dur = Duration::nanoseconds(d.to_nanos());
         let t = Utc::now()
             .with_timezone(tz)
             .date()
-            .and_hms(0, 0, 0)
+            .and_hms(0, 0, 0) // TODO: might panic!
```

Comment:
When would it panic?

Reply:
Can you use https://docs.rs/chrono/0.4.0/chrono/struct.Date.html#method.and_hms_opt and handle this somehow?

Reply:
Yes. However this will still fail the tests, because you will get `None` anyway, so it requires some more complicated change to fix. I would like to mark it as TODO and solve it in future.

Reply:
Ok.